tests/func/test_100_register.py

```diff
@@ -1,6 +1,15 @@
 import functools
 import pytest
 from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.support.ui import WebDriverWait
+
+
+def load_another_random(controller, reader_driver, ignore_list):
+    while True:
+        reader_driver.load_random()
```

Comment:
there is an optional argument to load_random, which is a `predicate` function which says "yes" or "no" using the `controller` object.
Sadly is it not used in every support module (ex: isekaiscan is not using it), but it should have been the proper way to handle these retries here.